File: JavaScript/LibaryApplication.js
const prompt = require('prompt-sync')();

const books = [
  { title: "Things Fall Apart - Chinua Achebe", available: true },
  { title: "Half of a Yellow Sun - Chimamanda Ngozi Adichie", available: true },
  { title: "Americanah - Chimamanda Ngozi Adichie", available: true },
  { title: "The Famished Road - Ben Okri", available: true },
  { title: "Purple Hibiscus - Chimamanda Ngozi Adichie", available: true },
  { title: "Cry, the Beloved Country - Alan Paton", available: true },
  { title: "Nervous Conditions - Tsitsi Dangarembga", available: true },
  { title: "Disgrace - J.M. Coetzee", available: true },
  { title: "The Joys of Motherhood - Buchi Emecheta", available: true },
  { title: "So Long a Letter - Mariama Bâ", available: true }
];

function showMenu() {
  console.log("\nLibrary Book Tracker");
  console.log("1. View all books");
  console.log("2. Borrow a book");
  console.log("3. Return a book");
  console.log("4. Exit");
}

function showBooks() { 
  console.log("\nBooks List:");
  for (let index = 0; index < books.length; index++) {
    const status = books[index].available ? "Available" : "Borrowed";
    console.log((index + 1) + ". " + books[index].title + " - " + status);
  }
}

function borrowBook() {
  showBooks();
  let input = prompt("Enter the number of the book to borrow: ");
  const count = Number(input) - 1;
  if (count >= 0 && count < books.length) {
    if (books[count].available) {
      books[count].available = false;
      console.log("You borrowed \"" + books[count].title + "\"");
    } else {
      console.log("Sorry, " + books[count].title + "has already borrowed.");
    }
  } else {
    console.log("Invalid book number.");
  }
}

function returnBook() {
  showBooks();
  let input = prompt("Enter the number of the book to return: ");
  const counter = Number(input) - 1;
  if (counter >= 0 && counter < books.length) {
    if (!books[counter].available) {
      books[counter].available = true;
      console.log("You returned \"" + books[counter].title + "\"");
    } else {
      console.log( books[counter].title + "Book was never borrowed.");
    }
  } else {
    console.log("Invalid book number.");
  }
}

function main() {
  while (true) {
    showMenu();
    let choice = prompt("Choose an option: ");
    if (choice === '1') {
      showBooks();
    } else if (choice === '2') {
      borrowBook();
    } else if (choice === '3') {
      returnBook();
    } else if (choice === '4') {
      console.log("Exiting program. Goodbye!");
      break;
    } else {
      console.log("Invalid option, please try again.");
    }
  }
}

main();

File: Python/libary_application.py
books = [
    { "title": "Things Fall Apart - Chinua Achebe", "available": True },
    { "title": "Half of a Yellow Sun - Chimamanda Ngozi Adichie", "available": True },
    { "title": "Americanah - Chimamanda Ngozi Adichie", "available": True },
    { "title": "The Famished Road - Ben Okri", "available": True },
    { "title": "Purple Hibiscus - Chimamanda Ngozi Adichie", "available": True },
    { "title": "Cry, the Beloved Country - Alan Paton", "available": True },
    { "title": "Nervous Conditions - Tsitsi Dangarembga", "available": True },
    { "title": "Disgrace - J.M. Coetzee", "available": True },
    { "title": "The Joys of Motherhood - Buchi Emecheta", "available": True },
    { "title": "So Long a Letter - Mariama Bâ", "available": True }
]

def is_numeric(string):
      for character in string:
        if character < '0' or character > '9':
            return False
    return len(string) > 0

def show_menu():
    print("\nLibrary Book Tracker")
    print("1. View all books")
    print("2. Borrow a book")
    print("3. Return a book")
    print("4. Exit")

def show_books():
    print("\nBooks List:")
    index = 0
    while index < len(books):
        if books[index]["available"]:
            status = "Available"
        else:
            status = "Borrowed"
        print(str(index + 1) + ". " + books[index]["title"] + " - " + status)
        index += 1

def borrow_book():
    show_books()
    user_input = input("Enter the number of the book to borrow: ")
    if is_numeric(user_input):
        num = int(user_input)
        index = num - 1
        if index >= 0 and index < len(books):
            if books[index]["available"]:
                books[index]["available"] = False
                print("You borrowed \"" + books[index]["title"] + "\"")
            else:
                print("Sorry, \"" + books[index]["title"] + "\" has already been borrowed.")
        else:
            print("Invalid book number.")
    else:
        print("Please enter a valid number.")

def return_book():
    show_books()
    user_input = input("Enter the number of the book to return: ")
    if is_numeric(user_input):
        num = int(user_input)
        index = num - 1
        if index >= 0 and index < len(books):
            if not books[index]["available"]:
                books[index]["available"] = True
                print("You returned \"" + books[index]["title"] + "\"")
            else:
                print("\"" + books[index]["title"] + "\" was never borrowed.")
        else:
            print("Invalid book number.")
    else:
        print("Please enter a valid number.")

def main():
    while True:
        show_menu()
        choice = input("Choose an option: ")
        if choice == "1":
            show_books()
        elif choice == "2":
            borrow_book()
        elif choice == "3":
            return_book()
        elif choice == "4":
            print("Exiting program. Goodbye!")
            break
        else:
            print("Invalid option, please try again.")

main()
